pkg/objectwait/rte/rte.go

```diff
@@ -94,9 +114,64 @@ func Deletable(mf rtemf.Manifests, cli client.Client, log logr.Logger) []objectw
 	}
 	if mf.MachineConfig != nil {
 		objs = append(objs, objectwait.WaitableObject{
-			// TODO: we should add functionality to wait for the MCP update
 			Obj: mf.MachineConfig,
+			Wait: func(ctx context.Context) error {
+				mcps, err := getMPCsForMC(cli, ctx, *mf.MachineConfig)
+				if err != nil {
+					return err
+				}
+				pollInterval := 30 * time.Second
+				pollTimeout := 30 * time.Minute
+				err = waitForMachineConfigPoolsCondition(cli, log, ctx, mcps, mcov1.MachineConfigPoolUpdating, pollInterval, pollInterval)
+				if err != nil {
+					return err
+				}
+				return waitForMachineConfigPoolsCondition(cli, log, ctx, mcps, mcov1.MachineConfigPoolUpdated, pollInterval, pollTimeout)
+			},
 		})
 	}
 	return objs
 }
+
+func waitForMachineConfigPoolsCondition(cli client.Client, log logr.Logger, ctx context.Context, mcps []*mcov1.MachineConfigPool, cond mcov1.MachineConfigPoolConditionType, interval, timeout time.Duration) error {
+	c1 := make(chan error)
+	for _, mcp := range mcps {
+		go func(mcp *mcov1.MachineConfigPool) {
+			err := wait.With(cli, log).
+				Interval(interval).
+				Timeout(timeout).
+				ForMachineConfigPoolCondition(ctx, mcp, cond)
+			c1 <- err
+		}(mcp)
+	}
+	var errs multierrors.MultiErrors
```

Comment:
it seems to me that we need at least 1 error to make the wait fail. IOW, it takes any error to fail. If so we should just return the first one got, while logging all of them. This can save us the addition of `multierror`